### DISCUSSION TITLE
Connection Timeout configuration

### DISCUSSION THREAD

**Discussion by iamcrisb** [orientechnologies/orientdb#9856] [Q&A] [answered]
Hi, our team is currently trying to configure the maximum time that a connection can be kept open and we are having difficulties with that.

Right now we are using the default configuration file and if we open a connection and not close it then it will be kept alive forever.

Can someone point me to the documentation where it says how we can configure timeouts in the context that I mentioned above?

Thank you!
 
Orientdb: v.2.2.36 (build d3beb772c02098ceaea89779a7afd4b7305d3788, branch 2.2.x)
Machine: Ubuntu 18

**Comment by tglman** [comment, marked as answer]
Hi,

2.2.x series is out of support for years now, the documentation is still online here: http://orientdb.org/docs/2.2.x/ but we strongly suggests you to move to a more recent version of OrientDB.

Regards

**Reply by iamcrisb** [reply]
Thank you @tglman! We got a legacy system on our hands so a migration is somehow really hard to put on the table right now. Is there a migration path that we could follow?

**Reply by tglman** [reply]
Hi,
We do have a strong compatibility in terms of APIs and data, you should be able to upgrade your code to a recent version like 3.0.x series (even though i would strongly suggest to go to the 3.2.x series) without any major change, the suggested path is updating your application code with the minor changes required cross version, than export the database from the old version and import it to a recent version and then point the update version to the new instance of the server.
